Inc/HALAL/Services/Communication/I2C/I2C.hpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * UART.hpp
+ *
+ *  Created on: 12 dec. 2022
+ *      Author: Pablo
+ */
+#pragma once
+
+#include "PinModel/Pin.hpp"
+#include "C++Utilities/CppUtils.hpp"
+#include "Packets/I2CPacket.hpp"
+#include "ErrorHandler/ErrorHandler.hpp"
+#include "DMA/DMA.hpp"
+
+#define HALAL_I2C_EMPTY_ID (uint16_t)0
+
+#ifdef HAL_I2C_MODULE_ENABLED
+
+/**
+ * @brief I2C service class. Abstracts the use of the UART service of the HAL library.
+ * 
+ */
+class I2C{
+private:
+    /**
+     * @brief Struct which defines all data referring to I2C peripherals. It is
+     *        declared private in order to prevent unwanted use. Only 
+     *        predefined instances should be used.
+     *           
+     */
+    struct Instance{
+        Pin SCL; /**< Clock pin. */
+        Pin SDA; /**< MOSI pin. */
+        I2C_HandleTypeDef* hi2c;  /**< HAL UART struct. */
+        I2C_TypeDef* instance;
+        DMA::Stream RX_DMA;
+        DMA::Stream TX_DMA;
+        uint8_t address = 0;
+        uint32_t speed_frequency = 100; /**< frequency in khz*/
+        uint32_t data_length = 8; /**< size in bits*/
+        bool receive_ready = false; /**< Receive value is ready to use pin. */
```

Comment:
```suggestion
        bool is_receive_ready = false; /**< Receive value is ready to use pin. */
```
no entiendo el comentario, que significa el pin?

Reply:
done